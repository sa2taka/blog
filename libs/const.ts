export const BLOG_TITLE = '園児ニアの庭園';
export const BLOG_DESCRIPTION = '庭いじり 遠目で見れば 引きこもり';
export const CTF_CATEGORY_ID = 'category';
export const CTF_POST_ID = 'blogPost';
