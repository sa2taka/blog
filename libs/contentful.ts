import createClient from '@/plugins/contentful';
import { MultipleItem, Post } from '@/types/entry';

const client = createClient();

export function fetchCategories() {
  return client.getEntries({
    content_type: process.env.CTF_CATEGORY_ID,
    order: 'fields.sort',
  });
}

export function fetchPosts(page: number, limit: number) {
  return client.getEntries({
    content_type: process.env.CTF_POST_ID,
    order: '-fields.releaseDate',
    skip: page * limit,
    limit,
  });
}

export function fetchLatestPostInCategory(category: string) {
  return client
    .getEntries({
      content_type: process.env.CTF_POST_ID,
      limit: 1,
      order: '-fields.releaseDate',
      'fields.category.sys.contentType.sys.id': process.env.CTF_CATEGORY_ID,
      'fields.category.fields.name': category,
    })
    .then((posts: MultipleItem<Post>) => posts.items[0]);
}

export function fetchPost(slug: string) {
  return client
    .getEntries({
      content_type: process.env.CTF_POST_ID,
      'fields.slug': slug,
    })
    .then((posts: MultipleItem<Post>) => posts.items[0]);
}
