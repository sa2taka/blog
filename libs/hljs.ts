// import hljs from 'highlight.js'; はバンドルサイズがやばい
// @ts-ignore
import hljs from 'highlight.js/lib/highlight';
import 'highlight.js/styles/atom-one-dark.css';

// @ts-ignore
import javascript from 'highlight.js/lib/languages/javascript';
// @ts-ignore
import typescript from 'highlight.js/lib/languages/typescript';
// @ts-ignore
import ruby from 'highlight.js/lib/languages/ruby';
// @ts-ignore
import xml from 'highlight.js/lib/languages/xml';
// @ts-ignore
import css from 'highlight.js/lib/languages/css';

hljs.registerLanguage('javascript', javascript);
hljs.registerLanguage('typescript', typescript);
hljs.registerLanguage('ruby', ruby);
hljs.registerLanguage('xml', xml);
hljs.registerLanguage('css', css);

export default hljs;
