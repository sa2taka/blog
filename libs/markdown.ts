import MarkdownIt from 'markdown-it';
// @ts-ignore
import footnote from 'markdown-it-footnote';
// @ts-ignore
import imsize from 'markdown-it-imsize';
// @ts-ignore
import katex from '@iktakahiro/markdown-it-katex';

import prism from '@/libs/prism';

const myHeaderPlugin = (md: MarkdownIt) => {
  md.renderer.rules.heading_open = (...[tokens, idx, options, _, self]) => {
    const tag = tokens[idx].tag;
    const content = tokens[idx + 1].content;
    const m = tag.match(/^h([123456])$/);
    if (m) {
      const headerIndex = Number(m[1]);
      const newIndex = headerIndex < 6 ? headerIndex + 1 : 6;
      return `<h${newIndex} id="${content}">`;
    } else {
      return self.renderToken(tokens, idx, options);
    }
  };
};

const myInlineCodePlugin = (md: MarkdownIt) => {
  md.renderer.rules.code_inline = (...[tokens, idx]) => {
    const tag = tokens[idx].tag;
    const content = tokens[idx].content;
    return `<${tag} class="hljsspan">${content}</${tag}>`;
  };
};

const myCodePlugin = (md: MarkdownIt) => {
  // code_blockではなくfence ruleにてcodeは拾われるらしい
  // @ts-ignore
  md.renderer.rules.fence = (...[tokens, idx]) => {
    const info = tokens[idx].info;
    const code = tokens[idx].content;

    let lang = '';
    let filename = '';
    if (info) {
      [lang, filename] = info.split(':').map((s) => s.trim());
    } else {
      lang = 'plaintext';
    }

    prism.languages.bash.prompt = /^[$#] /m;

    let value: string;
    if (prism.languages[lang]) {
      value = prism.highlight(code, prism.languages[lang], lang);
    } else {
      value = code;
      // shellで使われるtoken($, #, (%))を選択させないように変更
      if (lang === 'plaintext' || lang === '' || lang.includes('sh')) {
        value = value.replace(
          /^([$#&]\s*)/gm,
          '<span class="token prompt">$1</span>'
        );
      }
      lang = '';
    }

    let fileElement = '';
    if (filename && filename !== '') {
      fileElement = `<div class="filename">${filename}</div>`;
    }
    return `${fileElement}<code class="${
      lang !== '' ? `language-${lang}` : ''
    } ${
      filename && filename !== '' ? 'padding-for-filename' : ''
    }">${value}</code>`;
  };
};

const myImgPlugin = (md: MarkdownIt) => {
  const defaultRender =
    md.renderer.rules.image ||
    function (tokens, idx, options, _, self) {
      return self.renderToken(tokens, idx, options);
    };
  md.renderer.rules.image = (...[tokens, idx, options, env, self]) => {
    tokens[idx].attrPush(['loading', 'lazy']);
    return defaultRender(tokens, idx, options, env, self);
  };
};

const myWebpConvertPlugin = (md: MarkdownIt) => {
  const imgRender =
    md.renderer.rules.image ||
    function (tokens, idx, options, _, self) {
      return self.renderToken(tokens, idx, options);
    };
  md.renderer.rules.image = (...[tokens, idx, options, env, self]) => {
    const src = tokens[idx].attrs?.filter((elm) => elm[0] === 'src')[0]?.[1];
    if (!src) {
      return imgRender(tokens, idx, options, env, self);
    }

    const m = src.match(/\/\/images\.ctfassets\.net/);
    if (!m) {
      return imgRender(tokens, idx, options, env, self);
    }

    const token = src.includes('?') ? '&' : '?';
    const webp = `${src}${token}fm=webp`;
    const imgTag = imgRender(tokens, idx, options, env, self);
    const webpTag = `<source srcset="${webp}" type="image/webp"/>`;
    return `<picture>${webpTag}${imgTag}</picture>`;
  };
};
export const markdown = new MarkdownIt({
  html: true,
  linkify: true,
  breaks: true,
  typographer: true,
  langPrefix: '',
})
  .use(myCodePlugin)
  .use(footnote)
  .use(imsize, { autofill: true })
  .use(myHeaderPlugin)
  .use(myInlineCodePlugin)
  .use(myWebpConvertPlugin)
  .use(myImgPlugin);

export function addKatex(md: MarkdownIt) {
  return md.use(katex, { throwOnError: false, errorColor: ' #cc0000' });
}
