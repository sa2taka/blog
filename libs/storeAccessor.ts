import { Store } from 'vuex';
import { getModule } from 'vuex-module-decorators';
import Categories from '~/store/categories';

let categoriesStore: Categories;

function initialiseStores(store: Store<any>): void {
  categoriesStore = getModule(Categories, store);
}

export { initialiseStores, categoriesStore };
