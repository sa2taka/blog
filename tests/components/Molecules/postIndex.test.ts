import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import PostIndex from '../../../components/Molecules/postIndex.vue';
import { posts } from '../../__mocks__/postData';
import { generateIndexies } from '@/libs/generateIndexies';

const localVue = createLocalVue();

jest.mock('contentful', () => ({
  createClient: (_: any) => {},
}));

describe('molecules/postIndex/index', () => {
  let vuetify: any;
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const mountWithVuetify = (options?: any) => {
    return mount(PostIndex, {
      localVue,
      vuetify,
      ...options,
    });
  };

  test('should render', () => {
    const markdown = posts[0].fields.body;
    const index = generateIndexies(markdown);
    const wrapper = mountWithVuetify({
      propsData: {
        index,
      },
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
