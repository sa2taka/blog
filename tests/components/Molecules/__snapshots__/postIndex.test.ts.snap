// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/postIndex/index should render 1`] = `
<nav class="post-index">
  <div class="post-index_header"><span class="post-index_title">目次</span></div>
  <ol class="my-4 post-index_content">
    <li><a href="#テストです" class="index-link">テストです</a>
      <ul class="ml-5">
        <li><a href="#テストです" class="index-link">テストです</a></li>
      </ul>
    </li>
  </ol>
  <div class="post-index_footer"><span class="post-index_close-button"><v-btn icon="" aria-label="index close" class=""><v-icon>fa-chevron-up</v-icon></v-btn></span></div>
</nav>
`;
